DisplayInfo/DisplayInfo.cpp

```diff
@@ -146,26 +149,34 @@ namespace Plugin {
             displayInfo.Freegpuram = ram;
         }
 
-        if (_connectionProperties->IsAudioPassthrough(boolean) == Core::ERROR_NONE) {
-            displayInfo.Audiopassthrough = boolean;
+        bool status = false;
+        if (_connectionProperties->IsAudioPassthrough(status) == Core::ERROR_NONE) {
+            displayInfo.Audiopassthrough = status;
         }
-        if (_connectionProperties->Connected(boolean) == Core::ERROR_NONE) {
-            displayInfo.Connected = boolean;
+        status = false;
+        if (_connectionProperties->Connected(status) == Core::ERROR_NONE) {
+            displayInfo.Connected = status;
         }
+
+        uint32_t value = 0;
         if (_connectionProperties->Width(value) == Core::ERROR_NONE) {
             displayInfo.Width = value;
         }
+        value = 0;
         if (_connectionProperties->Height(value) == Core::ERROR_NONE) {
             displayInfo.Height = value;
         }
 
-        if (static_cast<const Exchange::IConnectionProperties*>(_connectionProperties)->HDCPProtection(hdcpType) == Core::ERROR_NONE) {
-            displayInfo.Hdcpprotection = static_cast<JsonData::DisplayInfo::DisplayinfoData::HdcpprotectionType>(hdcpType);
+        Exchange::IConnectionProperties::HDCPProtectionType hdcpProtection(Exchange::IConnectionProperties::HDCPProtectionType::HDCP_Unencrypted);
+        if ((const_cast<const Exchange::IConnectionProperties*>(_connectionProperties))->HDCPProtection(hdcpProtection) == Core::ERROR_NONE) {
```

Comment:
const_cast by definition is a scry thing. I think you are now invoking the wrong method here. You probably now invoke the SET while you where looking for a get, I assume.
Looking at the sugnature, https://github.com/rdkcentral/ThunderInterfaces/blob/master/interfaces/IDisplayInfo.h#L111 the method is const so in a const method, const objects will be used (const Exchange::IConnectionProperties* _connectionProperties) which should only give you access to the getter, without the const cast.
If you get a compiler error/warning, would love to see it :-)
Bottomline, I think the const_cast should go here

Reply:
Removed the const_cast. Compiler didn't throw any errors or warnings. I think we're good to go